src/picongpu/include/particles/ionization/byField/BSIStarkShifted/AlgorithmBSIStarkShifted.hpp

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2014-2016 Marco Garten
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "pmacc_types.hpp"
+#include "simulation_defines.hpp"
+#include "particles/traits/GetIonizationEnergies.hpp"
+#include "particles/traits/GetAtomicNumbers.hpp"
+#include "traits/attribute/GetChargeState.hpp"
+#include "algorithms/math/floatMath/floatingPoint.tpp"
+
+/** \file AlgorithmBSIStarkShifted.hpp
+ *
+ * IONIZATION ALGORITHM for the BSIStarkShifted model
+ *
+ * - implements the calculation of ionization probability and changes charge states
+ *   by decreasing the number of bound electrons
+ * - is called with the IONIZATION MODEL, specifically by setting the flag in @see speciesDefinition.param
+ */
+
+namespace picongpu
+{
+namespace particles
+{
+namespace ionization
+{
+
+    /** \struct AlgorithmBSIStarkShifted
+     *
+     * \brief calculation for the Barrier Suppression Ionization model
+     */
+    struct AlgorithmBSIStarkShifted
+    {
+
+        /** Functor implementation
+         *
+         * \tparam EType type of electric field
+         * \tparam BType type of magnetic field
+         * \tparam ParticleType type of particle to be ionized
+         *
+         * \param bField magnetic field value at t=0
```

Comment:
Ah, this is just a remnant from the pusher documentation. It means t=0 with respect to the current time step.
In the pushers you also sometimes get data from t=1/2 or t=-1/2.